# This patch changes Ruby's build system to always use baseruby (passed via --with-baseruby) as miniruby
# rather than building a miniruby executable, even when not cross-compiling.
# This fixes build errors if building Ruby with `-fsanitize=address` when not cross-compiling.

--- a/configure.ac
+++ b/configure.ac
@@ -3316,7 +3316,7 @@ for var in bindir includedir libdir rubylibprefix; do
 done
 
 BTESTRUBY='$(MINIRUBY)'
-AS_IF([test x"$cross_compiling" = xyes], [
+AS_IF([true], [
   test x"$MINIRUBY" = x && MINIRUBY="${RUBY-$BASERUBY} -I`$CHDIR .; pwd` "-r'$(arch)-fake'
   XRUBY_LIBDIR=`${RUBY-$BASERUBY} -rrbconfig -e ['puts RbConfig::CONFIG["libdir"]']`
   XRUBY_RUBYLIBDIR=`${RUBY-$BASERUBY} -rrbconfig -e ['puts RbConfig::CONFIG["rubylibdir"]']`
